kore/src/Kore/Builtin/Builtin.hs

```diff
@@ -516,10 +517,13 @@ data UnifyEq = UnifyEq
 
 -- | Unification of @eq@ symbols
 unifyEq ::
-    forall unifier.
+    forall unifier simplifier t.
+    MonadSimplify simplifier =>
     MonadUnify unifier =>
+    MonadTrans t =>
+    t simplifier ~ unifier =>
     TermSimplifier RewritingVariableName unifier ->
-    NotSimplifier unifier ->
+    NotSimplifier simplifier ->
```

Comment:
Can you explain a bit more how this change impacts the UnifierT issue?

As far as I can tell, we're putting less constraints on the time param sent to `NotSimplifier`, and we're imposing the monad transformer structure on the `unifier` param.

Reply:
Since #3380, the only source of nested `UnifierT` has been `notSimplifier` of type `MonadSimplify m => NotSimplifier (UnifierT m)` , since its ends up calling 
https://github.com/runtimeverification/haskell-backend/blob/324773459647d80f8af74f1d4cba66930f9e7796/kore/src/Kore/Simplify/And.hs#L220-L230
Here the `simplifier` variable was `UnifierT m` so `termAndWorker` had 2 `UnifierT`s (the outer one was removed by `runUnifierT` and inner one belongs to `notSimplifier` which calls this function).
Thus, removing the `MonadUnifier` constraint from type variable of `NotSimplifier`  allows us to remove `UnifierT` from `notSimplifier` passed as an argument to `runUnifierT` .